### PR TITLE
Use `node16` module resolution in typescript

### PR DESCRIPTION
**Pull request by timotheeguerin**
This makes it that typescript now understand that we need `.js` extension for imports

A little issue with this is there seems to be some problems with many `commonjs` libraries not having correct types and this new resolution system doesn't like it.

### CODE REVIEW

File: packages/cadl-vscode/package.json [hunk 1]
```diff
@@ -21,7 +21,7 @@
   "categories": [
     "Programming Languages"
   ],
-  "type": "commonjs",
+  "type": "module",
```
Comment (nguerrera) [inline packages/cadl-vscode/package.json:24]: oh, this works now for the vscode extension! Hooray. I think there are some workarounds that could be removed.

Comment (timotheeguerin) [reply packages/cadl-vscode/package.json:24]: not really, this just says that we wrote the code with modules but it still gets compiled to commonjs in rollup.js

Comment (timotheeguerin) [reply packages/cadl-vscode/package.json:24]: let me also double check everything works fine with the extension here

Comment (nguerrera) [reply packages/cadl-vscode/package.json:24]: OK, check F5 too where we run extension without rollup

Comment (timotheeguerin) [reply packages/cadl-vscode/package.json:24]: though maybe it also just work with module, let me try that too

Comment (timotheeguerin) [reply packages/cadl-vscode/package.json:24]: yeah unfortunately it's still not compatible with modules

**Review by nguerrera — APPROVED** [review]

**Pull request merged**